Light Vox Engine/Time.cpp

```diff
@@ -0,0 +1,56 @@
+#include "Time.h"
+
+Time* Time::instance = nullptr;
+
+Time * Time::GetInstance()
+{
+	if (instance == nullptr)
+		instance = new Time();
+
+	return instance;
+}
+
+void Time::ReleaseInstance()
+{
+	if(instance != nullptr)
+		delete instance;
+}
+
+void Time::UpdateTimer()
+{
+	__int64 now;
+	QueryPerformanceCounter((LARGE_INTEGER*)&now);
+	currentTime = now;
+
+	//Calc delta time and clamp to zero
+	//Could go negitive if CPU goes into power saver mode
+	//of the process itself gets moves to another core
+	deltaTime = max((float)((currentTime - previousTime) * perfCounterSeconds), 0.0f);
+	totalTime = (float)((currentTime - startTime) * perfCounterSeconds);
```

Comment:
This looks like a lot of pipeline flushings with the conversion between `__int64` to `double` to `float` -- maybe we can change currentTime, previousTime, and startTime to be floats and we can cast from now (`__int64`) to currentTime (`float`)